src/Expr.js

```diff
@@ -169,35 +228,114 @@ var exprToString = function(expr, caller) {
 
   if ('object' in expr) return printObject(expr['object'])
 
+  if ('select' in expr) {
+    return (
+      'Select(' +
+      exprToString(expr['select']) +
+      ', ' +
+      exprToString(expr['from']) +
+      ')'
+    )
+  }
+
+  if ('containsstr' in expr) {
+    return (
+      'ContainsStr(' +
+      exprToString(expr['containsstr']) +
+      ', ' +
+      exprToString(expr['search']) +
+      ')'
+    )
+  }
+
+  if ('filter' in expr) {
+    return (
+      'Filter(' +
+      exprToString(expr['collection']) +
+      ', ' +
+      exprToString(expr['filter']) +
+      ')'
+    )
+  }
+
+  if ('lambda' in expr) {
+    return (
+      'Lambda(' +
+      exprToString(expr['lambda']) +
+      ', ' +
+      exprToString(expr['expr']) +
+      ')'
+    )
+  }
+
+  if ('foreach' in expr) {
+    return (
+      'Foreach(' +
+      exprToString(expr['collection']) +
+      ', ' +
+      exprToString(expr['foreach']) +
+      ')'
+    )
+  }
+
+  if ('if' in expr) {
+    return (
+      'If(' +
+      exprToString(expr['if']) +
+      ', ' +
+      exprToString(expr['then']) +
+      ', ' +
+      exprToString(expr['else']) +
+      ')'
+    )
+  }
+
+  if ('call' in expr) {
+    var argumentExpr = exprToString(expr['arguments'])
+
+    // Print when arguments are an object
+    if (
+      typeof expr['arguments'] === 'object' &&
+      !Array.isArray(expr['arguments']) &&
+      !expr['arguments']['raw']
+    ) {
+      argumentExpr = printObject(expr['arguments'])
+    }
+
+    return 'Call(' + exprToString(expr['call']) + ', ' + argumentExpr + ')'
+  }
+
   // Versioned queries/lambdas will have an api_version field.
-  // We want to prevent it from being parsed and displayed as:
-  // Query(ApiVersion("3", "X", Var("X")))
+  // We want to prevent it from being parsed and displayed.
   var keys = Object.keys(expr).filter(
     expression => expression !== 'api_version'
   )
+
   var fn = keys[0]
 
-  // For FQL functions with special formatting concerns, we
-  // use the specialCases object above to define their casing.
-  if (fn in specialCases) fn = specialCases[fn]
+  // Handle functions with an arity greater than 1
+  if (keys.length > 1) {
+    fn = getFunctionFromKeys(keys)
+  }
 
-  fn = fn
-    .split('_')
-    .map(function(str) {
-      return str.charAt(0).toUpperCase() + str.slice(1)
-    })
-    .join('')
+  // If undefined, we may have an arguments object
+  if (fn === undefined) {
+    return keys
+  }
+
+  fn = convertToCamelCase(fn)
 
   var args = keys.map(function(k) {
     var v = expr[k]
     return exprToString(v, fn)
   })
 
-  var shouldReverseArgs = ['filter', 'map', 'foreach'].some(function(fn) {
-    return fn in expr
-  })
+  // Return string without args to avoid showing null args
+  if (args.length === 1 && args[0] === 'null') {
+    return fn + '()'
```

Comment:
this will cause incorrect behaviors, not all functions with 1-arity that receives `null` should be rendered as `fn()`, for example, `IsNumber(null)` as is will be rendered as `IsNumber()`

Reply:
Can you clarify this a bit more? My understanding was that whenever a function has one argument which is `null`, we don't want to render that.

Reply:
only 0-arity functions should drop null, like `Now()`. `IsNumber(x)` is 1-arity function and whatever argument was passed in should be preserved. You need to look at the wire to protocol to understand better, given `{}` represents a function call and the fields represents the function/arguments name, `Now()` function becomes `{"now": null}` in the wire, the null value is there because something must be in the value. On the other hand, `IsNumber(x)` is a 1-arity function, and it becomes `{"is_number": x}` in the wire